openshift_performance/ose3_perf/scripts/build_test_1.py

```diff
@@ -0,0 +1,414 @@
+        logger.error("Command failed:  tproject=" + namespace + ",cmd=" +
+                     err.cmd + ", retcode=" + str(err.returncode) + ", output="
+                     + err.output)
+    except Exception as e:
+        logger.error("cannot get build_name from: " + build_name)
+        logger.error(e)
+
+
+def check_build_status(executor):
+    logger.info("check_build_status ...")
+    # wait the first build is started
+    time.sleep(20)
+    while not all_builds_completed():
+        try:
+            time.sleep(10)
+            result = run("oc get build --all-namespaces --no-headers")
+            parse(executor, result)
+        except Exception as e:
+            logger.error(e)
+
+
+def all_builds_completed():
+    logger.debug("global_build_status: " + str(global_build_status))
+    for key, value in global_build_status.iteritems():
+        if value < 300:
+            return False
+    return True
+
+
+def parse(executor, result):
+    for line in result.splitlines():
+        words = line.split()
+
+        if len(words) >= 5:
+            status = words[4]
+            namespace = words[0]
+            name = words[1]
+            idx = namespace + ":" + name
+            if (status.startswith("Failed")) or (status == "Cancelled") or \
+                    (status.startswith("Error")):
+                if idx in global_build_status.keys():
+                    logger.debug(idx + " FAILED")
+                    if global_build_status[idx] < STATUS_NOT_COMPLETE:
+                        global_build_status[idx] = STATUS_NOT_COMPLETE
+                        stats_idx = idx[0:idx.rindex('-')]
+                        global_build_stats[stats_idx]["failed"] += 1
+            elif "Complete" == words[4]:
+                if idx in global_build_status.keys():
+                    if global_build_status[idx] < STATUS_COMPLETE:
+                        logger.info(idx + " Complete")
+                        global_build_status[idx] = STATUS_COMPLETE
+                    duration_string = words[-1]
+                    if global_build_status[idx] < STATUS_LOGGING:
+                        executor.submit(do_post_actions, namespace, name,
+                                        timeparse(duration_string))
+        else:
+            logger.error("unexpected return "
+                         "(oc get build --all-namespaces --no-headers): "
+                         + result)
+
+
+def do_post_actions(namespace, build_name, build_time):
+    idx = namespace + ":" + build_name
+    try:
+        if (global_build_status[idx] >= STATUS_LOGGING) or \
+                (global_build_status[idx] < STATUS_COMPLETE):
+            return
+        global_build_status[idx] = STATUS_LOGGING
+        start_regex = re.compile(
+            ".*(\d\d\d\d-\d\d-\d\dT\d\d:\d\d:\d\d).\d+Z\sPushing")
+        end_regex = re.compile(
+            ".*(\d\d\d\d-\d\d-\d\dT\d\d:\d\d:\d\d).\d+Z\s(Successfully pushed|"
+            + "Push successful)")
```

Comment:
Follow up issue to be created to no compile the regex for every post action.   We can either pass the compiled regex in or just do a re.match().   Not a perf sensitive part of the code in any case.

Reply:
Will do with another commit.